docs/syntax/Cadence.syntax

```diff
@@ -0,0 +1,1482 @@
+        access eventKeyword nonReservedIdentifier parameterList
+    ;
+
+comma:
+        ","
+    ;
+
+pragmaDeclaration:
+        "#" expression
+    ;
+
+entitlementDeclaration:
+        access entitlementKeyword nonReservedIdentifier
+    ;
+
+entitlementMappingDeclaration:
+        access entitlementKeyword mappingKeyword nonReservedIdentifier "{" entitlementMappingsAndInclusions "}"
+    ;
+
+entitlementMappingsAndInclusions:
+        *entitlementMappingElement
+    ;
+
+entitlementMappingElement:
+        entitlementMapping
+    |   entitlementIncludeType
+    ;
+
+entitlementMapping:
+        entitlementMappingSource "->" entitlementMappingTarget
+    ;
+
+entitlementMappingSource:
+        fullType
+    ;
+
+entitlementMappingTarget:
+        fullType
+    ;
+
+entitlementIncludeType:
+        includeKeyword nominalType
+    ;
+
+semicolonEnumerationDeclarationCase:
+        semicolon
+    |   enumerationDeclarationCase
+    ;
+
+enumerationDeclarationCase:
+        access caseKeyword nonReservedIdentifier
+    ;
+
+parameterList:
+        "(" *parameter( "," ) *1comma ")"
+    ;
+
+parameter:
+        parseLabeledParameter
+    |   parseUnlabeledParameter
+    ;
+
+parseLabeledParameter:
+        parameterPublicName parameterLocalName ":" typeAnnotation *1defaultValue
+    ;
+
+parseUnlabeledParameter:
+        parameterLocalName ":" typeAnnotation *1defaultValue
+    ;
+
+parameterPublicName:
+        nonReservedIdentifier
+    ;
+
+parameterLocalName:
+        nonReservedIdentifier
+    ;
+
+defaultValue:
+        "=" expression
+    ;
+
+typeAnnotation:
+        *1resourceAnnotation fullType
+    ;
+
+resourceAnnotation:
+        "@"
+    ;
+
+fullType:
+        fullTypeNormal
+    |   fullTypeNested
+    ;
+
+fullTypeNormal:
+        *1fullTypeReference innerType *fullTypeOptional
+    ;
+
+fullTypeReference:
+        *1fullTypeReferenceAuth "&"
+    ;
+
+fullTypeReferenceAuth:
+        fullTypeReferenceAuthConjunction
+    |   fullTypeReferenceAuthDisjunction
+    |   fullTypeReferenceAuthNoEntitlements
+    |   fullTypeReferenceAuthMapping
+    ;
+
+fullTypeReferenceAuthConjunction:
+        authKeyword "(" 1*fullType( "," ) ")"
+    ;
+
+fullTypeReferenceAuthDisjunction:
+        authKeyword "(" 1*fullType( "|" ) ")"
+    ;
+
+fullTypeReferenceAuthNoEntitlements:
+        authKeyword
+    ;
+
+fullTypeReferenceAuthMapping:
+        authKeyword "(" mappingKeyword nominalType ")"
+    ;
+
+fullTypeOptional:
+        . optional
+    ;
+
+fullTypeNested:
+        "(" fullType ")" *fullTypeOptional
+    ;
+
+innerType:
+        innerTypeBase
+    |   innerTypeRestrictions
+    ;
+
+innerTypeBase:
+        baseType *1innerTypeBaseRestrictions
+    ;
+
+innerTypeBaseRestrictions:
+        . typeRestrictions
+    ;
+
+innerTypeRestrictions:
+        typeRestrictions
+    ;
+
+baseType:
+        nominalType
+    |   functionType
+    |   variableSizedType
+    |   constantSizedType
+    |   dictionaryType
+    ;
+
+typeRestrictions:
```

Comment:
Restriction types got removed in Cadence 1.0. Instead, there are only intersection types, which have this syntax, but not as a "suffix" to a type. For example, before a type `T` could be restricted with `T{...}`, where `...` is a list of **zero** or more nominal types. Now there is just the syntax for intersection types `{...}` where `...` is a list of **one** or more nominal types.